src/cleanup-funcs.h

```diff
@@ -26,4 +28,20 @@
  **/
 void sc_cleanup_string(char **ptr);
 
+/**
+ * Close an open file.
+ *
+ * This function is designed to be used with
+ * __attribute__((cleanup(sc_cleanup_file))).
```

Comment:
I'd suggest defining an easy-to-type macro for "__attribute__ ((cleanup(sc_cleanup_endmntent)))". I think "autofclose" is descriptive.